numpy/core/src/multiarray/dragon4.c

```diff
@@ -1187,8 +1170,7 @@ Dragon4(const npy_uint64 mantissa, const npy_int32 exponent,
      * increases the number. This will either correct digitExponent to an
      * accurate value or it will clamp it above the accurate value.
      */
-    if ( (cutoffMode == CutoffMode_FractionLength ||
-          cutoffMode == CutoffMode_MaxFractionUnique) &&
+    if (cutoffNumber >= 0 && cutoffMode == CutoffMode_FractionLength &&
```

Comment:
Can you add a comment explaining the purpose of `cutoffNumber >= 0`? the only comment i can see in this diff about `cutoffNumber == -1` is a deletion!